server/server/src/serve_frontend.rs

```diff
@@ -0,0 +1,58 @@
+use actix_web::{get, http::header::ContentType, web::Data, HttpRequest, HttpResponse, Responder};
+use mime_guess::{from_path, mime};
+use rust_embed::RustEmbed;
+
+use crate::discovery::ServerInfo;
+
+#[derive(RustEmbed)]
+// Relative to server/Cargo.toml
+#[folder = "../../client/packages/host/dist"]
+struct Asset;
+
+// https://github.com/pyrossh/rust-embed/blob/master/examples/actix.rs
+fn server_frontend(path: &str) -> HttpResponse {
+    match Asset::get(path) {
+        Some(content) => HttpResponse::Ok()
+            .content_type(from_path(path).first_or_octet_stream().as_ref())
+            .body(content.data.into_owned()),
+        None => HttpResponse::NotFound().body("file not found"),
+    }
+}
+
+// Match file paths (ending  ($) with dot (\.) and at least one character (.+) )
+#[get(r#"/{filename:.*\..+$}"#)]
+async fn file(req: HttpRequest) -> impl Responder {
+    let filename: String = req.match_info().query("filename").parse().unwrap();
+    server_frontend(&filename)
+}
+
+// Match all paths
+#[get("/{_:.*}")]
+async fn index(_: HttpRequest) -> impl Responder {
+    server_frontend("index.html")
+}
+
+// Config js, to replace API_HOST with server IP
+// Ideally would use something like conforma f/e but capacitor mobile f/e build might fail to deduce server url from window.url
+// https://github.com/openmsupply/application-manager-web-app/blob/develop/src/config.ts
+#[get("/config.js")]
```

Comment:
probably not worth commenting.. as this is changing in another PR.
This gave me grief! Took me a while to realise that the site was running on the same port as graphql, so I tweaked my config to run the whole thing on port 443.. then had trouble because my certs are tied to `localhost` and this is using the IP. 

Have now generated a cert which supports both IP and localhost.. tweaked config and it's working nicely!

Reply:
btw in the downstream change `localhost` should work.